include/circt/Support/PrettyPrinterHelpers.h

```diff
@@ -166,6 +166,46 @@ class TokenStringSaver : public PrettyPrinter::Listener {
   void clear() override;
 };
 
+template <typename Callable, typename Data>
+class CallbackSaver {
```

Comment:
Neat that this is templated over the data type!

Reply:
Maybe this should be called `CallbackTokenGenerator` ? Since it doesn't really save callbacks-- they must already be saved, this saves a bounce callback forwarding through the stored data table using generated id's..?